compiler/luci/pass/src/SubstituteExpandDimsToReshapePass.cpp

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/SubstituteExpandDimsToReshapePass.h"
+
+#include <luci/IR/CircleNodes.h>
+#include <luci/Profile/CircleNodeOrigin.h>
+
+/**
+ * @brief Convert expand_dims op to reshape op
+ *        (expand_dims op with const axis ONLY)
+ * @example
+ *             input.shape = [2,3,4]
+ *             expand_dims(input, axis=1)
+ *
+ *             can be converted to
+ *
+ *             reshape(input, [2,1,3,4])
+ *
+ * BEFORE
+ *           |
+ *      [CircleNode]   [CircleConst]
+ *           \               /
+ *          [CircleExpandDims]
+ *                   |
+ *              [CircleNode]
+ *                   |
+ *
+ * AFTER
+ *           |
+ *      [CircleNode]  [CircleConst]
+ *           \              /
+ *           [CircleReshape]
+ *                   |
+ *              [CircleNode]
+ *                   |
+ */
+namespace
+{
+
+int32_t unknown_dim_count(luci::CircleNode *node)
+{
+  int32_t count = 0;
+
+  for (uint32_t i = 0; i < node->rank(); ++i)
+    if (!node->dim(i).known())
+      ++count;
+
+  return count;
+}
+
+/**
+ * @brief Return value in position on CircleConst with int32 format.
```

Comment:
```suggestion
 * @brief Return value in position on CircleConst with int32/int64 format.
```

Reply:
@jinevening It was to mean that it **returns** int32 value, so let me clarify that :-D

Reply:
Thanks!